protocol/src/protocol.calls.proposed.ts

```diff
@@ -0,0 +1,111 @@
+import { TextDocumentRegistrationOptions, StaticRegistrationOptions, TextDocumentPositionParams } from './protocol';
+
+export interface CallsClientCapabilities {
+    /**
+     * The text document client capabilities
+     */
+    textDocument?: {
+        /**
+         * Capabilities specific to the `textDocument/calls`
+         */
+        calls?: {
+            /**
+             * Whether implementation supports dynamic registration. If this is set to `true`
+             * the client supports the new `(TextDocumentRegistrationOptions & StaticRegistrationOptions)`
+             * return value for the corresponding server capability as well.
+             */
+            dynamicRegistration?: boolean;
+        };
+    }
+}
+
+export interface CallsServerCapabilities {
+    /**
+     * The server provides Call Hierarchy support.
+     */
+    callsProvider?: boolean | (TextDocumentRegistrationOptions & StaticRegistrationOptions);
+}
+
+/**
+ * A request to resolve all calls at a given text document position of a symbol definition or a call the same.
+ * The request's parameter is of type [CallsParams](#CallsParams), the response is of type [CallsResult](#CallsResult) or a
+ * Thenable that resolves to such.
+ *
+ *
+ */
+export namespace CallsRequest {
+    export const type = new RequestType<CallsParams, CallsResult, void, TextDocumentRegistrationOptions>('textDocument/calls');
+    export type HandlerSignature = RequestHandler<CallsParams, CallsResult | null, void>;
+}
+
+/**
+ * The parameters of a `textDocument/calls` request.
+ */
+export interface CallsParams extends TextDocumentPositionParams {
+    /**
+     * Outgoing direction for callees.
+     * The default is incoming for callers.
+     */
+    direction?: CallDirection;
+}
+
+/**
+ * Enum of call direction kinds
+ */
+export enum CallDirection {
+    /**
+     * Incoming calls aka. callers
+     */
+    Incoming = "incoming",
+    /**
+     * Outgoing calls aka. callees
+     */
+    Outgoing = "outgoing",
+}
+
+/**
+ * The result of a `textDocument/calls` request.
+ */
+export interface CallsResult {
+    /**
+     * The symbol of a definition for which the request was made.
+     *
+     * If no definition is found at a given text document position, the symbol is undefined.
+     */
+    symbol?: DocumentSymbol;
+    /**
+     * List of calls.
+     */
+    calls: Call[];
+}
+
+/**
+ * Represents a directed call.
+ */
+export interface Call {
+    /**
+     * Direction of the call.
+     */
+    direction: CallDirection;
+    /**
+     * Actual location of a call to a definition.
+     */
+    callLocation: Location;
+    /**
+     * Symbol refered to by this call. For outgoing calls this is a callee,
+     * otherwise a caller.
+     */
+    symbol: DocumentSymbol;
+    /**
+     * Location of the symbol, which can be used for a follow-up request to obtain the next level of calls.
+     */
+    symbolLocation: Location;
```

Comment:
Instead of adding the `symbolLocation` we should allow a `Location` instead of the `Range` within `DocumentSymbol`. As suggested by @kittaakos [here](https://github.com/Microsoft/vscode-languageserver-node/pull/346#discussion_r221659062).

Reply:
Unfortunately, that is a huge breaking change in the Java implementation of the LSP. :(